### PR TITLE
Populates fuchsia node actions in semantics updates.

### PR DESCRIPTION
**Pull request by abrush21**
## Description

This change ensures that the accessibility bridge correctly specifies which semantics actions are available to each node in a semantics update.

## Related Issues

https://bugs.fuchsia.dev/p/fuchsia/issues/detail?id=57668&q=owner%3Ame&can=2

## Tests

I added the following tests:

Added an accessibility bridge unit test verifying that semantics actions are translated correctly.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [X] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation.
- [X] All existing and new tests are passing.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/fuchsia/flutter/accessibility_bridge.cc [hunk 1]
```diff
@@ -115,6 +115,36 @@ fuchsia::accessibility::semantics::States AccessibilityBridge::GetNodeStates(
   return states;
 }
 
+std::vector<fuchsia::accessibility::semantics::Action>
+AccessibilityBridge::GetNodeActions(const flutter::SemanticsNode& node,
+                                    size_t* additional_size) const {
+  std::vector<fuchsia::accessibility::semantics::Action> node_actions;
+
+  if (node.HasAction(flutter::SemanticsAction::kTap)) {
+    node_actions.push_back(fuchsia::accessibility::semantics::Action::DEFAULT);
```
Comment (chunhtai) [inline shell/platform/fuchsia/flutter/accessibility_bridge.cc:124]: This is more a FYI. For a lack of a better name, we are thinking adding a semantics action that call DefaultAction to support ios magic tap. https://developer.apple.com/documentation/objectivec/nsobject/1615137-accessibilityperformmagictap?language=objc.

I am not sure how should they overlap here. Is the Action::DEFAULT always referring to tap in fuchsia?

File: shell/platform/fuchsia/flutter/accessibility_bridge.cc [hunk 2]
```diff
@@ -227,6 +257,7 @@ void AccessibilityBridge::AddSemanticsNodeUpdate(
         .set_transform(GetNodeTransform(flutter_node))
         .set_attributes(GetNodeAttributes(flutter_node, &this_node_size))
         .set_states(GetNodeStates(flutter_node, &this_node_size))
+        .set_actions(GetNodeActions(flutter_node, &this_node_size))
```
Comment (chunhtai) [inline shell/platform/fuchsia/flutter/accessibility_bridge.cc:260]: I am not familiar with fuchsia code. Do you know where are these APIs defined?

File: shell/platform/fuchsia/flutter/accessibility_bridge.cc (hunk 1 above)
Comment (abrush21) [reply shell/platform/fuchsia/flutter/accessibility_bridge.cc:124]: Thanks for the heads-up. Based on the description in that link, it looks like Action::Default is the most appropriate mapping for SemanticsAction::kTap.

File: shell/platform/fuchsia/flutter/accessibility_bridge.cc (hunk 2 above)
Comment (abrush21) [reply shell/platform/fuchsia/flutter/accessibility_bridge.cc:260]: Are you talking about the fuchsia node API? https://source.corp.google.com/piper///depot/google3/third_party/unsupported_toolchains/fuchsia/sdk/fidl/fuchsia_accessibility_semantics/node.fidl?q=fuchsia%20node%20fidl&ct=os&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental

Comment (chunhtai) [reply shell/platform/fuchsia/flutter/accessibility_bridge.cc:260]: ah I see, thanks for the link

**Pull request merged**